packages/core/database/src/entity-manager/regular-relations.ts

```diff
@@ -83,7 +80,11 @@ const deletePreviousOneToAnyRelations = async ({
     .delete()
     .from(joinTable.name)
     // Exclude the ids of the current document
-    .whereNotIn(joinColumn.name, getDocumentSiblingIdsQuery(con, joinColumn.referencedTable!, id))
+    .whereNotIn(
+      // @ts-expect-error - knex incorrectly expects a string array
+      joinColumn.name,
```

Comment:
@alexandrebodin  only when the second argument is a function, knex expects the first one to be an array of strings, do you know if that should be the case? This specific setup seems to work 🤷‍♂️  and the documentation doesn't mention anything about that either

Reply:
That's weird, I didn't have the issue in my PR :thinking:

Reply:
this only happened to me if the second argument was a function (qb) => qb.subquery
When using a regular subquery , it worked fine 😞

Reply:
super weird 🤔  

Another options is to build the query this where. not sure the typings would work better but who knows
```
 .where((builder) =>
    builder.whereIn('id', [1, 11, 15]).whereNotIn('id', [17, 19])
  )
```

Reply:
fixes the type issue I do see the original max stack size error 🙃